### ISSUE TITLE
Refinery doesn't obfuscate the Metrics API Key when logging errors

### ISSUE THREAD

**Issue by irvingpop** [honeycombio/refinery#522]
Refinery appears to log the configured metrics API key:

```
time="2022-09-20T14:36:35Z" level=error msg="Metrics reporter got an error back from Honeycomb" body= duration="176.703µs" error="Post \"https://api.honeycomb.io/1/batch/refinery-metrics\": net/http: invalid header field value \"<THE_API_KEY>\\n\" for key X-Honeycomb-Team" status_code=0
```

**Comment by kentquirk** [comment]
This was technically a bug in the Go language system libraries, where the http/net library had a bug where it was returning errors that contained the contents of unparseable headers. 

That bug was fixed in Go [in this change](https://github.com/golang/go/commit/e822b1e26e20ef1c76672c0b77b0fd8a97a1fe84), which was released as part of Go 1.19. Since we rebuilt our backend with Go 1.19 last week, this problem will not recur.
